cli/src/main/scala/com/codacy/analysis/cli/toolRepository/ToolRepositoryFactory.scala

```diff
@@ -21,7 +22,10 @@ object ToolRepositoryFactory {
 
       val toolsClient =
         ToolsClient(codacyApiBaseUrl)(httpClient = httpClient, ec = actorSystem.dispatcher, mat = materializer)
-      new ToolRepositoryRemote(toolsClient)(ec = actorSystem.dispatcher, mat = materializer)
+      val toolsDataStorage = new ToolSpecDataStorage()
+      new ToolRepositoryRemote(toolsClient, toolsDataStorage, PatternSpecDataStorage.apply)(
```

Comment:
Should we pass both data storage as lambdas? So it's a consistent approach

Reply:
You're right, it gets better by making it consistent.
Passing both as lambdas.